### ISSUE TITLE
Trying to access Google Search API through Webmaster on a server side app

### ISSUE THREAD

**Issue by bogdancss** [googleapis/google-api-nodejs-client#2501]
Hello,

I am trying to authorise `webmaster` requests to access data from the Google Search Console API using the Service Account keys.

I have created a service account, enabled G Suite Domain-wide Delegation and created JSON keys. Providing these keys to auth and then calling the webmaster query results in a 403 Forbidden error.

Here's the sample code I am trying to run:

```
const auth = new google.auth.GoogleAuth({
    keyFile: path.join(__dirname, './keys.json'),
    scopes: ['https://www.googleapis.com/auth/webmasters', 'https://www.googleapis.com/auth/webmasters.readonly']
  })
  google.options({ auth })

  const res = await webmasters.searchanalytics.query({
    siteUrl: 'https://www.hello.com',
    requestBody: {
      startDate: '2019-01-01',
      endDate: '2020-12-01'
    }
  })
```

As mentioned, the result is a 403 Forbidden error: `User does not have sufficient permission for site ...`
The account has full access to the site I am trying to access, so that is not an issue.

Any help would be appreciated.

**Comment by sofisl** [comment]
Hi @bogdancss,

In all honesty, I don't have a domain to reproduce the issue, but I'd like to see if we can solve the issue just by making sure the other details are set.

1. Have you confirmed you are using the most recent version of `googleapis`?
2. Are you using your oAuth2 credentials, which should look like:

```
{"web":{"client_id":"XXXXX.apps.googleusercontent.com","project_id":"PROJECT_ID","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"CLIENT_SECRET","redirect_uris":["http://localhost:3000/oauth2callback"]}}
```

You can get these in your console by going to `APIs & Services` > `Credentials` > `Create Credentials` > `OAuth Client ID`.

3. Can you use this code sample (I think it's basically the same as what you're already using, just want to type it out fully):

```

'use strict';

const path = require('path');
const {google} = require('googleapis');
const {authenticate} = require('@google-cloud/local-auth');

const webmasters = google.webmasters('v3');

async function runSample() {
  // Obtain user credentials to use for the request
  const auth = await authenticate({
    keyfilePath: OAUTH_CLIENT_ID_PATH,
    scopes: [
      'https://www.googleapis.com/auth/webmasters',
      'https://www.googleapis.com/auth/webmasters.readonly',
    ],
  });
  google.options({auth});

  const res = await webmasters.searchanalytics.query({
    siteUrl: 'SITE_URL',
    requestBody: {
      startDate: '2018-01-01',
      endDate: '2018-04-01',
    },
  });
  console.log(res.data);
  return res.data;
}

if (module === require.main) {
  runSample().catch(console.error);
}
module.exports = runSample;
```

4. Lastly, it might be helpful to look through [this article](https://support.google.com/webmasters/answer/7687615?visit_id=637461179015947834-3969224967&rd=2#zippy=%2Cadd-or-remove-users) discussing permissions.

Let me know if these tips don't work for you, and I'll give another shot at reproducing the issue.

**Comment by bogdancss** [comment]
hey @sofisl  - thanks for the instructions above.

I managed to find a solution by following the instructions on [this](https://stackoverflow.com/questions/36582143/oauth-2-0-server-to-server-credentials-authorization-failsearch-console-webma) SO issue posted over 4 years ago.

What I was trying to achieve was to authorise the requests to the Google Search API on the Node app without any user interaction, therefore I had to use a Google Service account. The issues is that there is no reference in any of the documentation that in order to use one of these accounts you will need to add the generated email address as an owner to the site in the Webmaster Tools frontend, in order to grant it access.

Hope this helps someone else.
